### ISSUE TITLE
Extra zero-length file created after 7.42.0 appears to have broken -z option

### ISSUE THREAD

**Issue by vszakats** [bagder/curl#237]
This code example (Windows platform):
```batch
echo TESTTEST> index.html
curl -fR -z .\index.html https://example.org/index.html -o .\index.html.newer
echo %ERRORLEVEL%
```
will create a _zero-length file_ and return result code 0 when using curl 7.42.0.

In previous versions it was returning result 0 but the zero-length file was _not_ created, which seems correct behavior, given that [-z](http://curl.haxx.se/docs/manpage.html#-z) means to download only if there is a newer than the reference-file found on the server. New behavior also makes it not possible to use `-z` either by checking the result code (always 0) or the existence of the output (always created).

This seems to be related to update https://github.com/bagder/curl/issues/183. Here, the description "curl: create output file on successful download of an empty file" (and Daniel's video) suggests that a zero-length file is only expected if the server actually serves a zero-length file, which is not the case in above scenario; the file on the server has a length of 1270 bytes.

Therefore I believe this to be a bug.

**Comment by bagder** [comment]
@kdudka, feel like looking into this?

**Comment by kdudka** [comment]
Sure, I will have a look.  Sorry for the breakage!

**Comment by kdudka** [comment]
Fixed in 710a2e99.  Thanks for the bug report!

**Comment by vszakats** [comment]
Thank you guys for the quick fix!
